fdbserver/KeyValueStoreMemory.actor.cpp

```diff
@@ -486,6 +568,10 @@ class KeyValueStoreMemory final : public IKeyValueStore, NonCopyable {
 								memset(&h, 0, sizeof(OpHeader));
 								memcpy(&h, data.begin(), data.size());
 								zeroFillSize = sizeof(OpHeader) - data.size() + h.len1 + h.len2 + 1;
+								if (h.op == OpEncrypted) {
+									// encryption header, plus the real (encrypted) op type
```

Comment:
nit: ASSERT(self->enableEncryption)?

Reply:
Makes sense. However, it would be better if we can guard it to `simulation` runs and not allow such a switch in production environments, imo.

Reply:
Then we are getting back to the previous discussion to store encryption state in cluster state..